src/annotator/anchoring/pdf.js

```diff
@@ -155,38 +156,45 @@ export async function documentHasText() {
 /**
  * Return the text of a given PDF page.
  *
+ * The page text returned by this function should match the `textContent` of the
+ * text layer element that PDF.js creates for rendered pages. This allows
+ * offsets computed in the text to be reused as offsets within the text layer
+ * element's content. This is important to create correct Ranges for anchored
+ * selectors.
+ *
  * @param {number} pageIndex
  * @return {Promise<string>}
  */
-async function getPageTextContent(pageIndex) {
+function getPageTextContent(pageIndex) {
+  // If we already have or are fetching the text for this page, return the
+  // existing result.
   if (pageTextCache[pageIndex]) {
```

Comment:
> This condition will always return true since this 'Promise<string>' is always defined.ts(2801)
pdf.js(171, 7): Did you forget to use 'await'?

This warning occurs when using TypeScript 4.4+. When I created this branch we were still using 4.3. I fixed the issue yesterday when merging a change from Dependabot that upgrades master to 4.4.